File: Programmers/포켓몬.cpp
#include <vector>
#include <algorithm>
using namespace std;

int solution(vector<int> nums)
{
    int n = nums.size() / 2;
    sort(nums.begin(), nums.end());
    for(int i = 1; i < nums.size(); i++)
        if(nums[i]==nums[i-1]) {
            nums.erase(nums.begin() + i);
            i--;
        }
    int answer = nums.size();
    while(answer > n)
        answer --;
    return answer;
}

File: Programmers/포켓몬.java
import java.util.Arrays;
import java.util.HashSet;
class Solution {
    public int solution(int[] nums) {
        int mx = nums.length/2;
        Arrays.sort(nums);
        int fs = nums[0];
        
        HashSet<Integer> set = new HashSet<>();
        for(int i = 0; i < nums.length; i++)
            set.add(nums[i]);
        
        if(mx<=set.size())
            return mx;
        else
            return set.size();
    }
}